### ISSUE TITLE
NullPointerException verifying schema in HiveClient

### ISSUE THREAD

**Issue by dain** [facebook/presto#1788]
The code that verifies the table schema matches the partition schema can throw a NPE at:
  https://github.com/facebook/presto/blob/0.76/presto-hive/src/main/java/com/facebook/presto/hive/HiveClient.java#L1068

One of the lists is likely null.

```
com.facebook.presto.spi.PrestoException: HIVE_UNKNOWN_ERROR
	at com.facebook.presto.hive.HiveSplitSourceProvider$HiveSplitSource.propagatePrestoException(HiveSplitSourceProvider.java:589) ~[na:na]
	at com.facebook.presto.hive.HiveSplitSourceProvider$HiveSplitSource.getNextBatch(HiveSplitSourceProvider.java:532) ~[na:na]
	at com.facebook.presto.execution.ConnectorAwareSplitSource.getNextBatch(ConnectorAwareSplitSource.java:46) ~[presto-main-0.76.jar:0.76]
	at com.facebook.presto.execution.SqlStageExecution.scheduleSourcePartitionedNodes(SqlStageExecution.java:633) [presto-main-0.76.jar:0.76]
	at com.facebook.presto.execution.SqlStageExecution.startTasks(SqlStageExecution.java:554) [presto-main-0.76.jar:0.76]
	at com.facebook.presto.execution.SqlStageExecution.access$200(SqlStageExecution.java:93) [presto-main-0.76.jar:0.76]
	at com.facebook.presto.execution.SqlStageExecution$4.run(SqlStageExecution.java:526) [presto-main-0.76.jar:0.76]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471) [na:1.7.0_51]
	at java.util.concurrent.FutureTask.run(FutureTask.java:262) [na:1.7.0_51]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_51]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_51]
	at java.lang.Thread.run(Thread.java:744) [na:1.7.0_51]
Caused by: java.lang.NullPointerException: null
	at com.facebook.presto.hive.HiveClient$1.apply(HiveClient.java:1068) ~[na:na]
	at com.facebook.presto.hive.HiveClient$1.apply(HiveClient.java:1039) ~[na:na]
	at com.google.common.collect.Iterators$8.transform(Iterators.java:794) ~[guava-16.0.1.jar:na]
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48) ~[guava-16.0.1.jar:na]
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48) ~[guava-16.0.1.jar:na]
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:543) ~[guava-16.0.1.jar:na]
	at com.facebook.presto.hive.HiveSplitSourceProvider.loadPartitionSplits(HiveSplitSourceProvider.java:190) ~[na:na]
	at com.facebook.presto.hive.HiveSplitSourceProvider.access$100(HiveSplitSourceProvider.java:80) ~[na:na]
	at com.facebook.presto.hive.HiveSplitSourceProvider$2.run(HiveSplitSourceProvider.java:172) ~[na:na]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471) [na:1.7.0_51]
	at java.util.concurrent.FutureTask.run(FutureTask.java:262) [na:1.7.0_51]
	at com.facebook.presto.hive.util.BoundedExecutor.executeOrMerge(BoundedExecutor.java:82) ~[na:na]
	at com.facebook.presto.hive.util.BoundedExecutor.access$000(BoundedExecutor.java:41) ~[na:na]
	at com.facebook.presto.hive.util.BoundedExecutor$1.run(BoundedExecutor.java:53) ~[na:na]
	... 3 common frames omitted
```

**Comment by dain** [comment]
The source of the error is now at:

https://github.com/facebook/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java#L445

**Comment by electrum** [comment]
Fixed in #2426
